gnd/src/main/java/com/google/android/gnd/persistence/local/room/RoomLocalDataStore.java

```diff
@@ -333,9 +335,29 @@ public Completable updateMutations(ImmutableList<Mutation> mutations) {
         .collect(toImmutableList());
   }
 
-  @Transaction
   @Override
-  public Completable removePendingMutations(ImmutableList<Mutation> mutations) {
+  public Completable finalizePendingMutations(ImmutableList<Mutation> mutations) {
+    return finalizeDeletions(mutations).andThen(removePending(mutations));
+  }
+
+  @Transaction
```

Comment:
`@Transaction` doesn't apply to Firebase, which uses it's own API (`db.runTransaction`) to batch writings. Please remove here and throughout.

Reply:
Done